server/main.go

```diff
@@ -52,7 +52,8 @@ func main() {
 	http.HandleFunc("/check/", CheckServer)
 	s := &http.Server{
 		Addr:        os.Args[1],
-		ReadTimeout: 10 * time.Second,
+		ReadTimeout: 1 * time.Second,
```

Comment:
Why change these? It's totally reasonable for a connection to stall for 1 second.